model/resolver/validation.go

```diff
@@ -54,6 +54,10 @@ func validateInstanceGroup(roleManifest *model.RoleManifest, g *model.InstanceGr
 			info.Name = name
 			jobReference.ResolvedConsumers[name] = info
 		}
+
+		if jobReference.ResolvedConsumedBy == nil {
```

Comment:
Why a validation function is mutating a parameter passed to it? I see the previous code is doing it already, but it is way out of its concern doing so. Could you add a TODO for moving it out, and create a card for it on our board?

Reply:
Added a card to tackle this.